cpr/session.cpp

```diff
@@ -272,6 +272,23 @@ void Session::SetUrl(const Url& url) {
     url_ = url;
 }
 
+void Session::SetResolve(const Resolve& resolve) {
+    SetResolves({resolve});
+}
+
+void Session::SetResolves(const std::vector<Resolve>& resolves) {
+    curl_slist_free_all(curl_->resolveCurlList_);
+    curl_->resolveCurlList_ = nullptr;
+    for (const Resolve& resolve : resolves) {
+        auto resolve_ports_iterator = resolve.ports.cbegin();
```

Comment:
Please avoid using `auto` when ever possible. In my eyes it makes code harder to understand.

Reply:
Any reason why this can't be a foreach loop?